cmsplugin_articles_ai/views.py

```diff
@@ -65,12 +67,40 @@ def get_context_data(self, **kwargs):
         context = super(ArticleListView, self).get_context_data(**kwargs)
         context.update({
             "all_tags": Tag.objects.all(),
+            "all_categories": Category.objects.all(),
             "page_title": self.tag_filter or _("All articles"),
             "tag_filter": self.tag_filter,
         })
         return context
 
 
+class CategoryView(DetailView):
```

Comment:
Couldn't you subclass this from `ArticleListView` since this should be pretty much a view with a list of articles. `ArticleListView` also provides tag filtering so you could filter a category with tags.